### PR TITLE
Refactored Add.js to use useFormik hook.

### PR DESCRIPTION
**Pull request by DDianovMD**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
-const queryClient = new QueryClient();
+// const queryClient = new QueryClient();
 
 function App() {
   return (
     <div className="App">
-      {/* <Router>
+      <Router>
         <NavBar />
         <Routes>
```
Comment (mariyan87) [inline src/App.js]: Router should have its own component

File: src/Components/add/Add.js [hunk 2]
```diff
@@ -1,72 +1,80 @@
...
+        phone: "",
+      },
+    },
+    onSubmit: (values) => {
+      axios
+        .post(URI, values.employee)
         .then((response) => {
-            document.location = 'http://localhost:3000'
-            alert('Successfully added employee!')
+          document.location = "http://localhost:3000";
```
Comment (mariyan87) [inline src/Components/add/Add.js]: use the router instead

File: src/Components/add/Add.js [hunk 3]
```diff
@@ -1,72 +1,80 @@
...
-        const updatedEmployee = {...employee, phone: inputValue}
-        setEmployee(updatedEmployee)
-    }
+const URI = `https://localhost:7189/api/employees`;
 
-    function handleAddButtonClick() {
-        axios.post(URI, employee)
+export function Add() {
+  const formik = useFormik({
+    initialValues: {
```
Comment (mariyan87) [inline src/Components/add/Add.js]: create a model and use it, when switching to ts, use its type

File: src/Components/add/Add.js [hunk 4]
```diff
@@ -1,72 +1,80 @@
...
-            </div>
-    )
-}
\ No newline at end of file
+  return (
+    <div className="d-flex justify-content-center flex-column">
+      <div style={{ alignSelf: "center" }}>
+        <h1>Add employee</h1>
+      </div>
+      <div style={{ width: "200px", alignSelf: "center" }}>
```
Comment (mariyan87) [inline src/Components/add/Add.js]: do not use inline styles, use scss

File: src/Components/add/Add.js [hunk 5]
```diff
@@ -1,72 +1,80 @@
...
-        setEmployee(updatedEmployee)
-    }
+const URI = `https://localhost:7189/api/employees`;
 
-    function handleAddButtonClick() {
-        axios.post(URI, employee)
+export function Add() {
+  const formik = useFormik({
+    initialValues: {
+      employee: {
```
Comment (mariyan87) [inline src/Components/add/Add.js]: the values should be only the model content, not  employee: {...}

**Review by mariyan87 — CHANGES_REQUESTED** [review]

File: src/Components/add/Add.tsx [hunk 6]
```diff
@@ -5,15 +5,23 @@ import { useNavigate } from "react-router-dom";
 
 const URI = `https://localhost:7189/api/employees`;
 
+type Employee = {
```
Comment (mariyan87) [inline src/Components/add/Add.tsx]: same as in c#, try to have single dedicated files for models and classes, etc. Create folder models and place the Employee in its own file

Comment (mariyan87) [reply src/Components/add/Add.tsx]: better change in to class and create constructor with empty fields for easy use. Similar to 
image

File: src/Components/add/Add.tsx [hunk 7]
```diff
@@ -5,15 +5,23 @@ import { useNavigate } from "react-router-dom";
...
   const formik = useFormik({
-    initialValues: {
-        firstName: "",
-        lastName: "",
-        phone: "",
-    },
+    initialValues: employee,
     onSubmit: (values) => {
       axios
         .post(URI, values)
```
Comment (mariyan87) [inline src/Components/add/Add.tsx]: create dedicated service file that is responsible for the given entity requests and have separate methods like createEmployee = (
  input: Employee
): Promise<Employee> => {
  return axios.post(BASE_URL, ....);
};

File: src/Components/edit/Edit.tsx [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+  lastName: string;
+  phone: string;
+};
+
+export function Edit(): React.JSX.Element {
+  const { state } = useLocation();
+  const navigate = useNavigate();
+
+  const id = state.id;
+  const URI = `https://localhost:7189/api/employees/${id}`;
```
Comment (mariyan87) [inline src/Components/edit/Edit.tsx]: Integrate https://www.npmjs.com/package/dotenv

File: src/Components/edit/Edit.tsx [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+  const formik = useFormik<Employee>({
+    initialValues: employee,
+
+    onSubmit: (values) => {
+      axios
+        .put(URI, values)
+        .then((response) => {
+          if (response.status === 204) {
+            alert("Successfully updated employee!");
```
Comment (mariyan87) [inline src/Components/edit/Edit.tsx]: alert is not user friendly. Use https://primereact.org/toast/ instead

File: src/Components/edit/Edit.tsx [hunk 10]
```diff
@@ -0,0 +1,101 @@
+import axios from "axios";
```
Comment (mariyan87) [inline src/Components/edit/Edit.tsx]: I wonder why you need separate add and edit components. The code is 90% the same.

File: src/Components/home/Home.tsx [hunk 11]
```diff
@@ -1,31 +1,41 @@
...
-    const [employees, setEmployees] = useState([{}])
-    
+type Employee = {
+    id: string,
+    firstName: string,
+    lastName: string,
+    phone: string,
+}
+
+export function Home(): React.JSX.Element {
```
Comment (mariyan87) [inline src/Components/home/Home.tsx]: rename to EmployeeList or something similar

File: src/Components/home/Home.tsx [hunk 12]
```diff
@@ -1,31 +1,41 @@
...
+            navigate("/");
         })
         .catch((error) => console.log(error));
     }
 
     useEffect(() => {
         axios.get(URI)
             .then((response) => {
-                setEmployees(response.data)
+                if(typeof(response.data) === typeof(employees)) {
```
Comment (mariyan87) [inline src/Components/home/Home.tsx]: Nice condition ;) what if is false ? show some error message, or so ?

File: src/Components/home/Home.tsx [hunk 13]
```diff
@@ -1,31 +1,41 @@
...
+                    setEmployees(response.data)
+                }
             })
             .catch((err) => console.log(err));
-    }, [])
+    }, [employees])
     
-    if(employees.length === 1) {
+    if(employees.length === 0) {
```
Comment (mariyan87) [inline src/Components/home/Home.tsx]: what if there are no records in the db ? you will always get 'loading'. Integrate useMutation from https://tanstack.com/query/latest/docs/react/reference/useMutation It give the ability to check the request state if is loading, failed, pending ..

**Review by mariyan87 — CHANGES_REQUESTED** [review]

**Comment by mariyan87** [comment]
@JoroPaspalev added to review and comment

File: src/Components/home/Home.tsx (hunk 13 above)
Comment (DDianovMD) [reply src/Components/home/Home.tsx]: Fixed.

File: src/Components/home/Home.tsx (hunk 12 above)
Comment (DDianovMD) [reply src/Components/home/Home.tsx]: Done.

File: src/Components/edit/Edit.tsx (hunk 9 above)
Comment (DDianovMD) [reply src/Components/edit/Edit.tsx]: Done.

File: src/Components/prime-react-homepage/PrimeReactHomePage.tsx [hunk 14]
```diff
@@ -0,0 +1,358 @@
...
+        })
+        .catch((error) => console.log(error));
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["getEmployees"] });
+    },
+  });
+
+  // Skeleton set
+  const items: DataTableValue[] = Array.from(
```
Comment (mariyan-marinov) [inline src/Components/prime-react-homepage/PrimeReactHomePage.tsx:150]: just name the variable in a better way, then no need of the comment above

File: src/Components/prime-react-homepage/PrimeReactHomePage.tsx [hunk 15]
```diff
@@ -0,0 +1,358 @@
...
+    });
+  };
+
+  const showInfoMessage = () => {
+    toast.current?.show({
+        severity: "info",
+        summary: "No changes were made.",
+        detail: "In order to update information you should change field values.",
+      });
+  }
```
Comment (mariyan-marinov) [inline src/Components/prime-react-homepage/PrimeReactHomePage.tsx]: these can be moved to a dedicated service

**Review by mariyan87 — APPROVED** [review]

**Review by mariyan-marinov — CHANGES_REQUESTED** [review]

File: src/Components/edit/Edit.tsx [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+  const URI = `https://localhost:7189/api/employees/${id}`;
+
+  const navigate = useNavigate();
+
+  const toast = useRef<Toast>(null);
+
+  const getEmployee = useQuery({
+    queryKey: ["getEmployee"],
+    queryFn: () => {
+      axios
```
Comment (mariyan-marinov) [inline src/Components/edit/Edit.tsx:21]: I still do not see seggragation here for these requests.
What would be: 1. Components only to call mutations.
2. All mutation declaration to be extracted into dedicated file(s). Every mutation refer mutationFn: someFunctionWithAxios
3. all request are extracted into dedicated file(s)

**Review by mariyan-marinov — CHANGES_REQUESTED** [review]
I still do not see seggragation here for these requests.
What would be: 1. Components only to call mutations.
2. All mutation declaration to be extracted into dedicated file(s). Every mutation refer mutationFn: someFunctionWithAxios
3. all request are extracted into dedicated file(s) - apicalls.ts

**Pull request merged**